src/htmx.js

```diff
@@ -3205,19 +3205,33 @@ var htmx = (function() {
     path = path || location.pathname + location.search
     const cached = getCachedHistory(path)
     if (cached) {
-      const fragment = makeFragment(cached.content)
-      const historyElement = getHistoryElement()
-      const settleInfo = makeSettleInfo(historyElement)
-      handleTitle(cached.title)
-      handlePreservedElements(fragment)
-      swapInnerHTML(historyElement, fragment, settleInfo)
-      restorePreservedElements()
-      settleImmediately(settleInfo.tasks)
-      getWindow().setTimeout(function() {
-        window.scrollTo(0, cached.scroll)
-      }, 0) // next 'tick', so browser has time to render layout
-      currentPathForHistory = path
-      triggerEvent(getDocument().body, 'htmx:historyRestore', { path, item: cached })
+      let restoreHistoryFromCache = function() {
+        const fragment = makeFragment(cached.content)
+        const historyElement = getHistoryElement()
+        const settleInfo = makeSettleInfo(historyElement)
+        handleTitle(cached.title)
+        handlePreservedElements(fragment)
+        swapInnerHTML(historyElement, fragment, settleInfo)
+        restorePreservedElements()
+        settleImmediately(settleInfo.tasks)
+        getWindow().setTimeout(function() {
+          window.scrollTo(0, cached.scroll)
+        }, 0) // next 'tick', so browser has time to render layout
+        currentPathForHistory = path
+        triggerEvent(getDocument().body, 'htmx:historyRestore', { path, item: cached })
+      }
+      const shouldTransition = htmx.config.globalViewTransitions
+      if (shouldTransition &&
+        // @ts-ignore experimental feature atm
+        document.startViewTransition) {
+        const innerRestoreHistoryFromCache = restoreHistoryFromCache
+        // wrap the original doRestoreHistory() in a call to startViewTransition()
+        // @ts-ignore experimental feature atm
+        restoreHistoryFromCache = document.startViewTransition(
+          () => innerRestoreHistoryFromCache()
+        )
+      }
```

Comment:
Since there is no potential delay here I think this be an if/else instead of wrapping the restoreHistoryFromCache function: pass it directly to startViewTransition else invoke it